### PR TITLE
[DOC] Add guide for deploying on Slurm

### PR DESCRIPTION
**Pull request by ddxxdd-code**
As mentioned by Issue #751 that the given link in the current docs for using Alpa on Ray on Slurm is broken. Hence, after using Slurm for one month, I wrote this guide, hoping to add to the tutorial section of docs and help people with access of Slurm to deploy Alpa.

### CODE REVIEW

File: docs/tutorials/alpa_on_slurm.rst [hunk 1]
```diff
@@ -0,0 +1,320 @@
+Using Alpa on Slurm
+##########################
+
+This page provides instructions to run Alpa on clusters managed by Slurm.
+This guide is modified from `Deploy Ray on Slurm <https://docs.ray.io/en/latest/cluster/vms/user-guides/community/slurm.html>`_.
+
+Prerequisites
+**************
+
+Alpa requires CUDA and runs on GPU. Therefore, you need access to CUDA-compatible GPU in the cluster to run Alpa.
```
Comment (zhisbug) [inline docs/tutorials/alpa_on_slurm.rst:10]: nits: GPUs

File: docs/tutorials/alpa_on_slurm.rst [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+Please make sure you have or can create an environment with required dependencies installed for Alpa.
+These dependencies are described in `Install Alpa <https://alpa.ai/install.html>`_.
+
+    .. note::
+
+        One easy way to create an environment with all dependencies installed is to create an environment using package management systems like `conda <https://docs.conda.io/en/latest/>`_ and setup the environment running an interactive job on Slurm.
```
Comment (zhisbug) [inline docs/tutorials/alpa_on_slurm.rst:17]: add a section to describe how to prepare an alpa python environment on slurm (assuming slurm has a Python virutalenv manager such as conda)

File: docs/tutorials/alpa_on_slurm.rst [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+        The interact job on Slurm can allow you to run and experiment in the exact way you do locally.
+        To run a interactive job on Slurm, use the command :code:`interact`.
```
Comment (zhisbug) [inline docs/tutorials/alpa_on_slurm.rst:19]: once you have a section in the previous comment, you can delete this note

File: docs/tutorials/alpa_on_slurm.rst [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+Create :code:`sbatch` Script
+****************************
+
+Usually large jobs like Alpa is run through sbatch on Slurm using a :code:`sbatch` script. :code:`sbatch` scripts are bash scripts with :code:`sbatch` options specified using :code:`#SBATCH <options>`.
+The Slurm cluster takes sbatch scripts submitted using command sbatch then queue the job specified by the script for execution.
```
Comment (zhisbug) [inline docs/tutorials/alpa_on_slurm.rst:25]: nits: queues

File: docs/tutorials/alpa_on_slurm.rst [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+
+Load Dependencies
+=================
+
+The next step is to setup the environment with Alpa's dependencies installed.
+In some Slurm clusters, CUDA, NVCC, and CuDNN are packed in containers that can be loaded directly. Here, we provide an example that loads a combination of available container and user-defined environment from package management systems.
+To load directly from available containers, use :code:`module load <module>`:
+
+    .. code:: bash
+
```
Comment (zhisbug) [inline docs/tutorials/alpa_on_slurm.rst:66]: add a note that if some modules are missing you have to find the slurm cluster manager to prepare them.

Also, add a note about how to check if a module exists.

File: docs/tutorials/alpa_on_slurm.rst [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+    .. note::
+
+        At this step, usually one can specify the version used if available like:
+
+        .. code:: bash
+
+            module load cuda/11.1.1
+            module load cudnn/8.0.5
+
+To activate a environment using package management systems like conda, add the following line:
```
Comment (zhisbug) [inline docs/tutorials/alpa_on_slurm.rst:80]: nits: an

File: docs/tutorials/alpa_on_slurm.rst [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+        ip_head=$head_node_ip:$port
+        export ip_head
+        # Start HEAD in background of head node
+        srun --nodes=1 --ntasks=1 -w "$head_node" \
+                ray start --head --node-ip-address="$head_node_ip" --port=$port \
+                --num-cpus "${SLURM_CPUS_PER_TASK}" --num-gpus $gpus_per_node --block &
+
+    .. note::
+
+        Note here the argument gpus_per_node should not exceed the number of GPU you have on each node.
```
Comment (zhisbug) [inline docs/tutorials/alpa_on_slurm.rst:151]: highlight

File: docs/tutorials/alpa_on_slurm.rst [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+        for ((i = 1; i <= worker_num; i++)); do
+            node_i=${nodes_array[$i]}
+            echo "Starting WORKER $i at $node_i"
+            srun --nodes=1 --ntasks=1 -w "$node_i" \
+                ray start --address "$ip_head" --num-cpus "${SLURM_CPUS_PER_TASK}" \
+                --num-gpus $gpus_per_node --block &
+            sleep 5
+        done
+
+At this time, we have a running Ray instance and next we can run Alpa on it.
```
Comment (zhisbug) [inline docs/tutorials/alpa_on_slurm.rst:170]: add a section to tell users to check ray is running as expected

**Review by zhisbug — CHANGES_REQUESTED** [review]

**Comment by zhisbug** [comment]
@ddxxdd-code please remove the `echo` statements in the scripts and only keep those important ones (e.g., echos that check ray is up and alpa is working fine)

**Comment by ddxxdd-code** [comment]
Just updated all the scripts. All echos removed, except for the one before testing Ray status.

**Review by zhisbug — CHANGES_REQUESTED** [review]

**Comment by zhisbug** [comment]
Also, please move the slurm_scripts_examples to under `examples/` instead of `docs/`

**Pull request merged**